finesse/hardware/plugins/temperature/senecat121.py

```diff
@@ -0,0 +1,141 @@
+"""This module provides an interface to Seneca temperature readers."""
+from decimal import Decimal
+from typing import Any
+
+import numpy
+from serial import SerialException
+
+from finesse.hardware.serial_device import SerialDevice
+
+from .temperature_monitor_base import TemperatureMonitorBase
+
+
+class SenecaError(Exception):
+    """Indicates that an error occurred while communicating with the device."""
+
+
+class SenecaT121(
+    SerialDevice,
+    TemperatureMonitorBase,
+    description="SenecaT121",
+    default_baudrate=57600,
+):
+    """An interface for Seneca T121 temperature readers.
+
+    The manual for this device is available at:
+    https://www.seneca.it/media/6848058/flyer_t12x_eng_nov_2020.pdf
+    """
+
+    def __init__(self, *serial_args: Any, **serial_kwargs: Any) -> None:
+        """Create a new Seneca from an existing serial device.
+
+        Args:
+            serial_args: Arguments to Serial constructor
+            serial_kwargs: Keyword arguments to Serial constructor
+        """
+        SerialDevice.__init__(self, *serial_args, **serial_kwargs)
+        TemperatureMonitorBase.__init__(self)
+
+    def close(self) -> None:
+        """Close the connection to the device."""
+        self.serial.close()
+
+    def read(self) -> bytes:
+        """Read temperature data from the Seneca.
+
+        Returns:
+            data: the sequence of bytes read from the device
+
+        Raises:
+            SenecaError: Malformed message received from device
+        """
+        try:
+            data = self.serial.read(size=21)
+        except SerialException as e:
+            raise SenecaError(e)
+
+        # require 21 bytes else checks will fail
+        min_length = 21
+        if len(data) != min_length:
+            raise SenecaError("Insufficient data read from device")
+
+        return data
+
+    def request_read(self) -> None:
+        """Write a message to the Seneca to prepare for a read operation.
+
+        Raises:
+            SenecaError: Error writing to the device
+        """
+        try:
+            self.serial.write(bytearray([1, 3, 0, 2, 0, 8, 229, 204]))
```

Comment:
Presumably this list of numbers means something specific, documented in the manual? Maybe it's worth adding a comment or something in the docstring to elucidate this.